### PR TITLE
Allow to filter for specific tasks

### PR DESCRIPTION
**Pull request by xoxys**
Supersedes: https://github.com/woodpecker-ci/autoscaler/pull/94

- use woodpecker-go
- add tests

### CODE REVIEW

**Comment by pat-s** [comment]
@xoxys Unblocked now with v2.5.0

**Comment by pat-s** [comment]
@anbraten You might wanna take a look?

**Review by anbraten — APPROVED** [review]

File: engine/autoscaler.go [hunk 1]
```diff
@@ -203,6 +195,27 @@ func (a *Autoscaler) cleanupAgents(ctx context.Context) error {
...
+		return queueInfo.Stats.Workers, queueInfo.Stats.Running, queueInfo.Stats.Pending, nil
+	}
+
+	labelFilterKey, labelFilterValue, ok := strings.Cut(a.config.FilterLabels, "=")
+	if !ok {
+		return 0, 0, 0, fmt.Errorf("invalid labels filter: %s", a.config.FilterLabels)
+	}
+
+	running := coundTasksByLabel(queueInfo.Running, labelFilterKey, labelFilterValue)
+	pending := coundTasksByLabel(queueInfo.Pending, labelFilterKey, labelFilterValue)
```
Comment (anbraten) [inline engine/autoscaler.go]:
```suggestion
	running := countTasksByLabel(queueInfo.Running, labelFilterKey, labelFilterValue)
	pending := countTasksByLabel(queueInfo.Pending, labelFilterKey, labelFilterValue)
```

File: config/config.go [hunk 2]
```diff
@@ -12,4 +12,5 @@ type Config struct {
 	GRPCAddress             string
 	GRPCSecure              bool
 	AgentAllowedStartupTime time.Duration
+	FilterLabels            string
```
Comment (anbraten) [inline config/config.go:15]: Just noticed this has to be set in main.go

Comment (xoxys) [reply config/config.go:15]: Good catch, fixed.

**Review by anbraten — APPROVED** [review]

**Pull request merged**